js/apps/admin-ui/src/identity-providers/add/DiscoverySettings.tsx

```diff
@@ -50,152 +33,92 @@ const Fields = ({ readOnly }: DiscoverySettingsProps) => {
-      </FormGroup>
-      <TextField
-        field="config.logoutUrl"
-        label="logoutUrl"
-        isReadOnly={readOnly}
+        type="url"
+        readOnly={readOnly}
+        rules={{
+          required: t("required"),
+        }}
+      />
+      <TextControl
+        name="config.logoutUrl"
+        label={t("logoutUrl")}
+        readOnly={readOnly}
       />
-      <TextField
-        field="config.userInfoUrl"
-        label="userInfoUrl"
-        isReadOnly={readOnly}
+      <TextControl
+        name="config.userInfoUrl"
+        label={t("userInfoUrl")}
+        readOnly={readOnly}
       />
-      <TextField field="config.issuer" label="issuer" isReadOnly={readOnly} />
-      <SwitchField
-        field="config.validateSignature"
-        label="validateSignature"
-        isReadOnly={readOnly}
+      <TextControl
+        name="config.issuer"
+        label={t("issuer")}
+        readOnly={readOnly}
+      />
+      <DefaultSwitchControl
+        name="config.validateSignature"
+        label={t("validateSignature")}
+        isDisabled={readOnly}
       />
       {validateSignature === "true" && (
         <>
-          <SwitchField
-            field="config.useJwksUrl"
-            label="useJwksUrl"
-            data-testid="useJwksUrl"
-            isReadOnly={readOnly}
+          <DefaultSwitchControl
+            name="config.useJwksUrl"
+            label={t("useJwksUrl")}
+            isDisabled={readOnly}
           />
           {useJwks === "true" ? (
-            <TextField
-              field="config.jwksUrl"
-              label="jwksUrl"
-              isReadOnly={readOnly}
+            <TextControl
+              name="config.jwksUrl"
+              label={t("jwksUrl")}
+              readOnly={readOnly}
             />
           ) : (
             <>
-              <FormGroupField label="validatingPublicKey">
-                <KeycloakTextArea
-                  data-testid="validatingPublicKey"
-                  aria-label={t("validatingPublicKey")}
-                  {...register("config.publicKeySignatureVerifier")}
-                />
-              </FormGroupField>
-              <TextField
-                field="config.publicKeySignatureVerifierKeyId"
-                label="validatingPublicKeyId"
-                isReadOnly={readOnly}
+              <TextControl
+                name="config.publicKeySignatureVerifier"
+                label="validatingPublicKey"
+              />
+              <TextControl
+                name="config.publicKeySignatureVerifierKeyId"
+                label={t("validatingPublicKeyId")}
+                readOnly={readOnly}
               />
             </>
           )}
         </>
       )}
-      <SwitchField
-        field="config.pkceEnabled"
-        label="pkceEnabled"
-        isReadOnly={readOnly}
+      <DefaultSwitchControl
+        name="config.pkceEnabled"
+        label={t("pkceEnabled")}
+        isDisabled={readOnly}
+        stringify
```

Comment:
If this gets `stringify`, should that not also happen for all the other places where `SwitchField` was replaced with `SwitchControl`? E.g. for `config.useJwksUrl`.